src/main/java/com/redhat/jenkins/plugins/cachet/matrix/CachetMatrixAction.java

```diff
@@ -0,0 +1,150 @@
+    public boolean isChecked(String combination) {
+        CachetJobProperty jobProperty = getCachetJobPropertyFromCombination(combination);
+        if (jobProperty != null) return jobProperty.getRequiredResources();
+        return false;
+    }
+
+    // For UI purposes - has the user selected this resource?
+    public boolean isSelected(String combination, String resource) {
+        return getCachetResources(combination).contains(resource);
+    }
+
+    // For UI purposes - get all the resources
+    public List<String> getResourceNames() {
+        return ResourceProvider.SINGLETON.getResourceNames();
+    }
+
+    /**
+     * @return all the possible combinations
+     */
+    public Iterable<Combination> getAxis() {
+        return project.getAxes().list();
+    }
+
+    /**
+     * Help method to avoid code duplication
+     * @param combination
+     * @return CachetJobProperty or null if not exist
+     */
+    private CachetJobProperty getCachetJobPropertyFromCombination(String combination){
+        MatrixConfiguration matrixConfiguration = project.getItem(combination);
+        assert matrixConfiguration != null;
+        return matrixConfiguration.getProperty(CachetJobProperty.class);
+    }
+
+    /**
+     * Help method to convert resources from JSONArray into List of Strings
+     * @param combination
+     * @param obj
+     * @return List of resources
+     */
+    private List<String> getResourcesFromCombination(Combination combination, JSONObject obj) {
+        List<String> resources = new ArrayList<>();
+        JSONObject resourcesObj = obj.getJSONObject(combination.toString());
+        JSONArray resourcesArray = (JSONArray) resourcesObj.get("resources");
+        resourcesArray.forEach(resource -> resources.add(resource.toString()));
+        return resources;
+    }
+
+    /**
+     * Get the right resources for some combination
+     * @param combination
+     * @return List of resources or empty list if there is no such JobProperty
+     * @throws NullPointerException
+     */
+    private List<String> getCachetResources(String combination)
+            throws NullPointerException {
+        CachetJobProperty jobProperty = getCachetJobPropertyFromCombination(combination);
+        if (jobProperty != null) return jobProperty.getResources();
+        else return Collections.emptyList();
+    }
+
+    /**
+     * Taking the resources for some combination from the UI into the correct configuration
+     * @param combination
+     * @param resources
+     * @throws IOException
+     */
+    protected void setCachetProperty(Combination combination, List<String> resources)
+            throws IOException {
+        MatrixConfiguration matrixConfiguration = project.getItem(combination);
+        matrixConfiguration.removeProperty(CachetJobProperty.class);
+        if (!resources.isEmpty())
+            matrixConfiguration.addProperty(new CachetJobProperty(true, resources));
+    }
+
+    // Handle Cachet configuration
```

Comment:
Do we still need that after we have switched to using axis?

Reply:
Yes, since I'm using it for the axis && in case you don't want to use the cachet-axis but want to set different resources (manually) to other kinds of axis (not likely but possible)